pubtools/pulplib/_impl/model/unit/erratum.py

```diff
@@ -374,6 +380,15 @@ class ErratumUnit(Unit):
     )
     """A list of package collections associated with the advisory."""
 
+    container_list = pulp_attrib(
+        type=list,
+        pulp_field="pulp_user_metadata.container_list",
+        converter=frozenlist_or_none_converter,
```

Comment:
There is a problem here, if the field contains a list of dict, it will be mutable.  frozenlist will be applied only at the object's top level, dicts within the list will be mutable.  All the fields need to be deeply immutable.

If you want it to be a complex structure with lists containing dicts containing lists and so on, please make sure there's a converter to make it all deeply immutable, i.e. all lists replaced with frozenlist and all dicts replaced with frozendict.

The existing frozendict_or_none_converter almost does this already, maybe some of that can be reused.